### ISSUE TITLE
iOS 8 position issues

### ISSUE THREAD

**Issue by batjo** [jmascia/KLCPopup#5]
Thanks for this amazing popup! I want to put in inside an app that I'm making but with iOS 8 around the corner I was wondering if you are going to fix position problems. Have a look at the screenshot.

Also the fade out animation flickers a bit. 

![ios simulator screen shot 13 aug 2014 10 49 56](https://cloud.githubusercontent.com/assets/3170237/3903053/de048f5a-22c6-11e4-97b0-a3c89bd21525.png)

**Comment by jmascia** [comment]
Hey @bartjochems, glad you like the popup! I'm sorry about the issues on iOS 8. I will look into it shortly. I think I should be able to fix it by this weekend...

**Comment by batjo** [comment]
That would be really great! Thanks a lot

**Comment by SamStone92** [comment]
Any updates for iOS 8?

**Comment by jmascia** [comment]
Did a lot of testing and figured out the issues. Just need to clean up my code and then I'll commit.

**Comment by SamStone92** [comment]
Amazing, thank you.

**Comment by frankcortes** [comment]
What is the current state for this issue? This bug is really annoying for the user experience. We are going to publish an app in the App Store and we would need to fix this before.

**Comment by batjo** [comment]
Hi @jmascia I did some more testing and found another problem when the orientation changes. Have a look at the attached screenshots: 
![ios simulator screen shot 10 sep 2014 14 44 50](https://cloud.githubusercontent.com/assets/3170237/4217499/68a0084c-38e8-11e4-83d3-34535ab12eab.png)
![ios simulator screen shot 10 sep 2014 14 44 40](https://cloud.githubusercontent.com/assets/3170237/4217500/6defca30-38e8-11e4-8ab2-554836622b24.png)

Looking forward to a new release!

**Comment by iremk** [comment]
any luck on fixing this issue?
looking forward to it.

**Comment by ancloid** [comment]
Quick fix orientation for iPad landscape only:

in KLCPopup.m search for :
self.transform = CGAffineTransformMakeRotation(angle);
self.frame = self.window.bounds;

change with:
if (floor(NSFoundationVersionNumber) <= NSFoundationVersionNumber_iOS_7_1) {
      self.transform = CGAffineTransformMakeRotation(angle);
      self.frame = self.window.bounds;
  }

**Comment by ristkari** [comment]
Wonderng whether there will be fix? It would be greatly appreciated :)

**Comment by ghost** [comment]
anclefeudor's fix for the iPad orientation issue did the trick for me. Many thanks and very nice control!

**Comment by batjo** [comment]
@jmascia any plans on releasing a new version?

**Comment by jmascia** [comment]
I put out a new version that should fix layout and rotation issues in iOS 8. Let me know if you find any issue, or give it a go yourself ;)

Apologies for the long wait on this. It's been a busy time both at home and work. Thanks for understanding.

**Comment by batjo** [comment]
Thanks a lot!

**Comment by petr-fiala** [comment]
great PopUp! definitely going to use it in my projects. There is still problem with orientations on iOS8... fix is super simple, already written in comments above, why won't you push it?

**Comment by fedemame** [comment]
@jmascia please merge the @anclefeudor fix, there is still a rotation bug in iOS 8.
Congratulation for the component anyway ;)

**Comment by khaledannajar** [comment]
Hi cocoapods always get version 1.0 and the fix is in version 1.1. Have you updated the cocoapods repository? @jmascia 

pod error:

```
[!] Unable to satisfy the following requirements:

- `KLCPopup (~> 1.1)` required by `Podfile`
```

**Comment by HarishSami** [comment]
When We Drag the KCLPopup.h and .m files to our project it cannot be working. Getting TypeNum redefinition Error. I can not write any code in .h and .m files. I am using Xcode 9.4.1. Is there ant Podfile for this KCLPopup
